Lib9c/Action/Sell4.cs

```diff
@@ -0,0 +1,147 @@
+            var states = ctx.PreviousStates;
+            if (ctx.Rehearsal)
+            {
+                states = states.SetState(ShopState.Address, MarkChanged);
+                states = states.SetState(sellerAvatarAddress, MarkChanged);
+                return states.SetState(ctx.Signer, MarkChanged);
+            }
+
+            var addressesHex = GetSignerAndOtherAddressesHex(context, sellerAvatarAddress);
+
+            var sw = new Stopwatch();
+            sw.Start();
+            var started = DateTimeOffset.UtcNow;
+            Log.Debug("{AddressesHex}Sell exec started", addressesHex);
+
+
+            if (price.Sign < 0)
+            {
+                throw new InvalidPriceException($"{addressesHex}Aborted as the price is less than zero: {price}.");
+            }
+
+            if (!states.TryGetAgentAvatarStates(ctx.Signer, sellerAvatarAddress, out AgentState agentState, out AvatarState avatarState))
+            {
+                throw new FailedLoadStateException($"{addressesHex}Aborted as the avatar state of the signer was failed to load.");
+            }
+            sw.Stop();
+            Log.Debug("{AddressesHex}Sell Get AgentAvatarStates: {Elapsed}", addressesHex, sw.Elapsed);
+            sw.Restart();
+
+            if (!avatarState.worldInformation.IsStageCleared(GameConfig.RequireClearedStageLevel.ActionsInShop))
+            {
+                avatarState.worldInformation.TryGetLastClearedStageId(out var current);
+                throw new NotEnoughClearedStageLevelException(addressesHex, GameConfig.RequireClearedStageLevel.ActionsInShop, current);
+            }
+
+            Log.Debug("{AddressesHex}Sell IsStageCleared: {Elapsed}", addressesHex, sw.Elapsed);
+
+            sw.Restart();
+            if (!states.TryGetState(ShopState.Address, out Bencodex.Types.Dictionary shopStateDict))
+            {
+                throw new FailedLoadStateException($"{addressesHex}Aborted as the shop state was failed to load.");
+            }
+
+            Log.Debug("{AddressesHex}Sell Get ShopState: {Elapsed}", addressesHex, sw.Elapsed);
+            sw.Restart();
+
+            Log.Debug("{AddressesHex}Execute Sell; seller: {SellerAvatarAddress}", addressesHex, sellerAvatarAddress);
+
+            // 인벤토리에서 판매할 아이템을 선택하고 수량을 조절한다.
+            if (!avatarState.inventory.TryGetNonFungibleItem(itemId, out ItemUsable nonFungibleItem))
+            {
+                throw new ItemDoesNotExistException(
+                    $"{addressesHex}Aborted as the NonFungibleItem ({itemId}) was failed to load from avatar's inventory."
+                );
+            }
+
+            if (nonFungibleItem.RequiredBlockIndex > context.BlockIndex)
+            {
+                throw new RequiredBlockIndexException(
+                    $"{addressesHex}Aborted as the equipment to enhance ({itemId}) is not available yet; it will be available at the block #{nonFungibleItem.RequiredBlockIndex}."
```

Comment:
```suggestion
                    $"{addressesHex}Aborted as the equipment to sell ({itemId}) is not available yet; it will be available at the block #{nonFungibleItem.RequiredBlockIndex}."
```